Adds SearchTest class to make sure the Search class works as intended.

BucketOfNubbins opened the pull request.

Commit:
Adds SearchTest class to make sure the Search class works as intended.

GregLeitheiser commented on src/test/java/org/servantscode/commons/search/SearchTest.java:
```diff
@@ -0,0 +1,223 @@
+
+    @Test
+    public void testSearchTextClause() {
+        Search search = new Search();
```
These can be broken up into several test cases.

GregLeitheiser commented on src/test/java/org/servantscode/commons/search/SearchTest.java:
```diff
@@ -0,0 +1,223 @@
+        assertEquals("Wrong SQL.", "field = ?", clause.getSql());
+        assertEquals("Wrong Values", "[-2147483648]", clause.getValues().toString());
+
+        for (int i = -100000; i < 100000; i++) {
```
Boom! Thorough but over kill. Hit a negative number, 0 and a positive number + the MIN and MAX.

GregLeitheiser commented on src/test/java/org/servantscode/commons/search/SearchTest.java:
```diff
@@ -0,0 +1,223 @@
+        assertEquals("Wrong Values", "[-2147483648, 2147483647]", clause.getValues().toString());
+
+        for (int i = -1000; i < 1000; i++) {
+            for (int j = -1000; j < 1000; j++) {
```
Ditto :)

GregLeitheiser reviewed (changes requested):
Please break up the tests into individual test cases (With good descriptive names) and trim down some of the long running tests.

Great first cut!

Commit:
Fixes issues with searchTest JUnit testing, shortening times and formatting

Commit:
Adds null parameter checks in the Search class, and fixes issues in the SearchTest JUnit testing class.
Now the SearchTest class checks for what happens when null values are passed, and expects exceptions to be thrown.
